validphys2/src/validphys/deltachi2.py

```diff
@@ -140,15 +140,12 @@ def pos_neg_xplotting_grids(delta_chi2_hessian, xplotting_grid):
     """
     positive_eigenvalue_mask = delta_chi2_hessian >= 0
 
-    # include replica 0 in both new grids
+    # The masks do not include replica 0, add it in both grids
     pos_mask = np.append(True, positive_eigenvalue_mask)
     neg_mask = np.append(True, ~positive_eigenvalue_mask)
 
-    pos_grid = xplotting_grid.grid_values.data[pos_mask]
-    neg_grid = xplotting_grid.grid_values.data[neg_mask]
-
-    pos_xplotting_grid = xplotting_grid.copy_grid(grid_values=pos_grid)
-    neg_xplotting_grid = xplotting_grid.copy_grid(grid_values=neg_grid)
+    pos_xplotting_grid = xplotting_grid.mask_replicas(pos_mask)
```

Comment:
Is this method expected to be used a lot? If this is going to be the only use I'd rather keep the functionality here than adding the method, mostly to not make xplotting_grid harder than it needs to be. cc @RoyStegeman

Reply:
> Is this method expected to be used a lot?

I guess no? But (answering your other question) then the `copy_grid` method needs to work with numpy arrays (or have an access to `__class__` here which is ugly).

Reply:
Probably not, the delta chi2 thing is sometimes used for students projects but I don't plan to do much with it myself. Maybe @scarlehoff  has other uses for mask_replica in mind?

Reply:
or wrap `pos_mask` in pdf.stats_class, no?